### PR TITLE
[FR] rust chapter8 continuation

### PR DESCRIPTION
**Pull request by mgemard**

### CODE REVIEW

File: lessons.yaml [hunk 1]
```diff
@@ -6065,6 +6065,33 @@ pages:
...
...r.add(1).write(14)%3B%0A%20%20%20%20%20%20%20%20%20%20%20%20ptr.add(2).write(73)%3B%0A%20%20%20%20%20%20%20%20%20%20%20%20ptr.add(3).write(64)%3B%0A%0A%20%20%20%20%20%20%20%20%20%20%20%20Pie%20%7B%20secret_recipe%3A%20ptr%20as%20usize%20%7D%0A%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%7D%0A%7D%0Aimpl%20Deref%20for%20Pie%20%7B%0A%20%20%20%20type%20Target%20%3D%20f32%3B%0A%20%20%20%20fn%20deref(%26self)%20-%3E%20%26f32%20%7B%0A%20%20%20%20%20%20%20%20%2F%2F%20Interpr%C3%A8te%20le%20pointer%20secret_recipe%20en%20tant%20que%0A%20%20%20%20%20%20%20%20%2F%2F%20pointer%20brut%20f32.%0A%20%20%20%20%20%20%20%20let%20pointer%20%3D%20self.secret_recipe%20as%20*const%20f32%3B%0A%20%20%20%20%20%20%20%20%0A%20%20%20%20%20%20%20%20%2F%2F%20D%C3%A9reference%20le%20pointer%20brut%20et%20retourne%20une%0A%20%20%20%20%20%20%20%20%2F%2F%20valeur%20%26f32.%0A%20%20%20%20%20%20%20%20unsafe%20%7B%20%26*pointer%20%7D%0A%20%20%20%20%7D%0A%7D%0Afn%20main()%20%7B%0A%20%20%20%20let%20p%20%3D%20Pie%3A%3Anew()%3B%0A%20%20%20%20%2F%2F%20On%20d%C3%A9r%C3%A9f%C3%A9rence%20le%20pointeur%20intelligent%20de%20notre%0A%20%20%20%20%2F%2F%20structure%20Pie.%0A%20%20%20%20println!(%22%7B%3A%3F%7D%22%2C%20*p)%3B%0A%7D%0A
```
Comment (VincentFoulon80) [inline lessons.yaml:6070]:
```suggestion
      code: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&code=use%20std%3A%3Aalloc%3A%3A%7Balloc%2C%20Layout%7D%3B%0Ause%20std%3A%3Aops%3A%3ADeref%3B%0A%0Astruct%20Pie%20%7B%0A%20%20%20%20secret_recipe%3A%20usize%2C%0A%7D%0A%0Aimpl%20Pie%20%7B%0A%20%20%20%20fn%20new()%20-%3E%20Self%20%7B%0A%20%20%20%20%20%20%20%20%2F%2F%20On%20d%C3%A9finit%20l'agencement%20que%20doit%20avoir%20la%20m%C3%A9moire%0A%20%20%20%20%20%20%20%20%2F%2F%20pour%20pouvoir%20allouer%204%20octets.%0A%20%20%20%20%20%20%20%20let%20layout%20%3D%20Layout%3A%3Afrom_size_align(4%2C%201).unwrap()%3B%0A%0A%20%20%20%20%20%20%20%20unsafe%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20%2F%2F%20Alloue%20nos%204%20octets%20en%20m%C3%A9moire%20et%20sauvegarde%20la%20position%0A%20%20%20%20%20%20%20%20%20%20%20%20%2F%2F%20de%20d%C3%A9but%20en%20tant%20qu'entier.%0A%20%20%20%20%20%20%20%20%20%20%20%20let%20ptr%20%3D%20alloc(layout)%20as%20*mut%20u8%3B%0A%20%20%20%20%20%20%20%20%20%20%20%20%0A%20%20%20%20%20%20%20%20%20%20%20%20%2F%2F%20On%20%C3%A9crit%20quelques%20valeurs%20u8%20en%20m%C3%A9moire%20avec%0A%20%20%20%20%20%20%20%20%20%20%20%20%2F%2F%20un%20peu%20d'arithm%C3%A9tique%20des%20pointeurs.%0A%20%20%20%20%20%20%20%20%20%20%20%20ptr.write(86)%3B%0A%20%20%20%20%20%20%20%20%20%20%20%20ptr.add(1).write(14)%3B%0A%20%20%20%20%20%20%20%20%20%20%20%20ptr.add(2).write(73)%3B%0A%20%20%20%20%20%20%20%20%20%20%20%20ptr.add(3).write(64)%3B%0A%0A%20%20%20%20%20%20%20%20%20%20%20%20Pie%20%7B%20secret_recipe%3A%20ptr%20as%20usize%20%7D%0A%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%7D%0A%7D%0Aimpl%20Deref%20for%20Pie%20%7B%0A%20%20%20%20type%20Target%20%3D%20f32%3B%0A%20%20%20%20fn%20deref(%26self)%20-%3E%20%26f32%20%7B%0A%20%20%20%20%20%20%20%20%2F%2F%20Interpr%C3%A8te%20le%20pointeur%20secret_recipe%20en%20tant%20que%0A%20%20%20%20%20%20%20%20%2F%2F%20pointeur%20brut%20f32.%0A%20%20%20%20%20%20%20%20let%20pointer%20%3D%20self.secret_recipe%20as%20*const%20f32%3B%0A%20%20%20%20%20%20%20%20%0A%20%20%20%20%20%20%20%20%2F%2F%20D%C3%A9r%C3%A9f%C3%A9rence%20le%20pointeur%20brut%20et%20retourne%20une%0A%20%20%20%20%20%20%20%20%2F%2F%20valeur%20%26f32.%0A%20%20%20%20%20%20%20%20unsafe%20%7B%20%26*pointer%20%7D%0A%20%20%20%20%7D%0A%7D%0Afn%20main()%20%7B%0A%20%20%20%20let%20p%20%3D%20Pie%3A%3Anew()%3B%0A%20%20%20%20%2F%2F%20On%20d%C3%A9r%C3%A9f%C3%A9rence%20le%20pointeur%20intelligent%20de%20notre%0A%20%20%20%20%2F%2F%20structure%20Pie.%0A%20%20%20%20println!(%22%7B%3A%3F%7D%22%2C%20*p)%3B%0A%7D%0A
```

File: lessons.yaml [hunk 2]
```diff
@@ -6065,6 +6065,33 @@ pages:
...
+      content_markdown: |
+        Revenons sur quelques pointeurs intelligents que tu connais déjà: `Vec<T>` et `String`.
+
+        `Vec <T>` est un pointeur intelligent vers des octets en mémoire. Le
+        compilateur Rust n'a aucune idée de la manière dont interpréter ces octets.
+        Le pointeur intelligent est capable d'interpréter ces octets en gardant
+        une trace de l'endroit dans ces octets où une structures de données commence
+        et se termine. Le pointeur intelligent déréférence un pointeur brut pour
+        exposer la structure de données dans une belle interface ergonomique propre
+        que nous d'utiliser (par exemple `my_vec [3]`).
```
Comment (VincentFoulon80) [inline lessons.yaml:6080]:
```suggestion
        `Vec <T>` est un pointeur intelligent vers des octets en mémoire. Le
        compilateur Rust n'a aucune idée de la manière dont il faut interpréter ces octets.
        Le pointeur intelligent est capable d'interpréter ces octets en gardant
        une trace de l'endroit dans ces octets où une structures de données commence
        et se termine. Le pointeur intelligent déréférence un pointeur brut pour
        exposer la structure de données dans une belle interface ergonomique propre
        que nous d'utiliser (par exemple `my_vec [3]`).
```

File: lessons.yaml [hunk 3]
```diff
@@ -6065,6 +6065,33 @@ pages:
...
+
+        De même, `String` garde une trace des octets en mémoire et restreint par
+        programmation le contenu qui y est écrit à toujours être de `utf-8`
+        valide. Il déréférence cette région de mémoire dans un type` &str`.
```
Comment (VincentFoulon80) [inline lessons.yaml:6084]:
```suggestion
        De même, `String` garde une trace des octets en mémoire et restreint par
        programmation le contenu qui y est écrit à toujours être de l'`utf-8`
        valide. Il déréférence cette région de mémoire dans un type` &str`.
```

File: lessons.yaml [hunk 4]
```diff
@@ -6065,6 +6065,33 @@ pages:
...
+
+        Ces deux structures de données doivent procéder à un déréférencement
+        dangereux de pointeurs bruts.
+
+        Détails de la mémoire:
+        * Rust possède l'équivalent de `malloc` du langage C avec
+        [alloc](https://doc.rust-lang.org/std/alloc/fn.alloc.html) et
+        [Layout](https://doc.rust-lang.org/std/alloc/struct.Layout.html)
+        pour allouer soi-même un région de la mémoire.
```
Comment (VincentFoulon80) [inline lessons.yaml:6093]:
```suggestion
        * Rust possède l'équivalent de `malloc` du langage C avec
        [alloc](https://doc.rust-lang.org/std/alloc/fn.alloc.html) et
        [Layout](https://doc.rust-lang.org/std/alloc/struct.Layout.html)
        pour allouer soi-même une région de la mémoire.
```

File: lessons.yaml [hunk 5]
```diff
@@ -6110,7 +6147,28 @@ pages:
         ```rust
         fn main() -> Result<(), Box<dyn std::error:Error>>
         ```
-        
+    fr:
+      title: Renvoie d'erreurs dans le main revisité
```
Comment (VincentFoulon80) [inline lessons.yaml:6151]:
```suggestion
      title: Renvoi d'erreurs dans le main revisité
```

File: lessons.yaml [hunk 6]
```diff
@@ -6135,6 +6193,17 @@ pages:
 
         Only when the last smart pointer is dropped does the data on the heap become deallocated.
         
+    fr:
+      title: Multi-Propriétaire
+      code: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&code=use%20std%3A%3Arc%3A%3ARc%3B%0A%0Astruct%20Pie%3B%0A%0Aimpl%20Pie%20%7B%0A%20%20%20%20fn%20eat(%26self)%20%7B%0A%20%20%20%20%20%20%20%20println!(%22C%27est%20bien%20meilleur%20sur%20le%20tas!%22)%0A%20%20%20%20%7D%0A%7D%0A%0Afn%20main()%20%7B%0A%20%20%20%20let%20heap_pie%20%3D%20Rc%3A%3Anew(Pie)%3B%0A%20%20%20%20let%20heap_pie2%20%3D%20heap_pie.clone()%3B%0A%20%20%20%20let%20heap_pie3%20%3D%20heap_pie2.clone()%3B%0A%0A%20%20%20%20heap_pie3.eat()%3B%0A%20%20%20%20heap_pie2.eat()%3B%0A%20%20%20%20heap_pie.eat()%3B%0A%0A%20%20%20%20%2F%2F%20Les%20donn%C3%A9es%20de%20Pie%20se%20trouvant%20sur%20le%20tas%20sont%0A%20%20%20%20%2F%2F%20d%C3%A9sallou%C3%A9es%20ici.%0A%7D%0A
+      content_markdown: |
+        `Rc` (en anglais Reference Counting) est un pointeur intelligent qui
+        déplace les données de la pile vers le tas. Celui-ci peut être cloner
+        et permet d'emprunter immuablement les données se trouvant sur le tas.
```
Comment (VincentFoulon80) [inline lessons.yaml:6202]:
```suggestion
        `Rc` (en anglais Reference Counting) est un pointeur intelligent qui
        déplace les données de la pile vers le tas. Celui-ci peut être cloné
        et permet d'emprunter immuablement les données se trouvant sur le tas.
```

**Comment by richardanaya** [comment]
deployed!

**Pull request merged**